src/sys/termios.rs

```diff
@@ -1,648 +1,782 @@
+                      not(any(target_arch = "powerpc", target_arch = "powerpc64")))))]
+        CIBAUD,
+        #[cfg(any(target_os = "android", target_os = "linux"))]
+        CBAUDEX,
+        #[cfg(any(target_os = "dragonfly",
+                  target_os = "freebsd",
+                  target_os = "macos",
+                  target_os = "netbsd",
+                  target_os = "openbsd"))]
+        MDMBUF,
+        #[cfg(any(target_os = "netbsd", target_os = "openbsd"))]
+        CHWFLOW,
+        #[cfg(any(target_os = "dragonfly",
+                  target_os = "freebsd",
+                  target_os = "netbsd",
+                  target_os = "openbsd"))]
+        CCTS_OFLOW,
+        #[cfg(any(target_os = "dragonfly",
+                  target_os = "freebsd",
+                  target_os = "netbsd",
+                  target_os = "openbsd"))]
+        CRTS_IFLOW,
+        #[cfg(any(target_os = "dragonfly",
+                  target_os = "freebsd"))]
+        CDTR_IFLOW,
+        #[cfg(any(target_os = "dragonfly",
+                  target_os = "freebsd"))]
+        CDSR_OFLOW,
+        #[cfg(any(target_os = "dragonfly",
+                  target_os = "freebsd"))]
+        CCAR_OFLOW,
+    }
+}
 
-        // XXX: We're using `repr(C)` because `c_int` doesn't work here.
-        // See https://github.com/rust-lang/rust/issues/10374.
-        #[derive(Clone, Copy)]
-        #[repr(C)]
-        pub enum FlushArg {
-            TCIFLUSH  = 0,
-            TCOFLUSH  = 1,
-            TCIOFLUSH = 2,
-        }
+libc_bitflags! {
+    /// Bitmasks for use with ControlFlags to select specific settings
+    pub flags ControlFlagsMask: tcflag_t {
+        CSIZE,
+    }
+}
 
-        // XXX: We're using `repr(C)` because `c_int` doesn't work here.
-        // See https://github.com/rust-lang/rust/issues/10374.
-        #[derive(Clone, Copy)]
-        #[repr(C)]
-        pub enum FlowArg {
-            TCOOFF = 0,
-            TCOON  = 1,
-            TCIOFF = 2,
-            TCION  = 3,
-        }
+libc_bitflags! {
+    /// Flags for setting any local modes
+    pub flags LocalFlags: tcflag_t {
+        ECHOKE,
+        ECHOE,
+        ECHOK,
+        ECHO,
+        ECHONL,
+        ECHOPRT,
+        ECHOCTL,
+        ISIG,
+        ICANON,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly",
+                  target_os = "macos",
+                  target_os = "netbsd",
+                  target_os = "openbsd"))]
+        ALTWERASE,
+        IEXTEN,
+        EXTPROC,
+        TOSTOP,
+        FLUSHO,
+        #[cfg(any(target_os = "freebsd",
+                  target_os = "dragonfly",
+                  target_os = "macos",
+                  target_os = "netbsd",
+                  target_os = "openbsd"))]
+        NOKERNINFO,
+        PENDIN,
+        NOFLSH,
     }
 }
 
+/// Get input baud rate (see
+/// [cfgetispeed(3p)](http://man7.org/linux/man-pages/man3/cfgetispeed.3p.html)).
```

Comment:
Since this function is standardized by POSIX, it would probably be better to link to the Open Group's man page, instead of Linux's.